<h1>ACTIVITY: Palmer Penguins Data Analytics</h1>
<hr>

<h4>Kyle M. De Los Santos</h4>

In [1]:
#l Libraries

In [2]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [3]:
bird = pd.read_csv("Datasets\\penguins.csv")

In [4]:
bird

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h2>Descriptive Analytics</h2>

In [6]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [7]:
bird.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Filtering a Column/Data</h3>

In [8]:
bird["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [9]:
bird["island"].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [10]:
adelie_df = bird[bird["species"] == "Adelie"]

In [11]:
adelie_df["body_mass_g"].mean()

np.float64(3700.662251655629)

In [12]:
gentoo_df = bird[bird["species"] == "Gentoo"]

In [13]:
gentoo_df["body_mass_g"].mean()

np.float64(5076.016260162602)

In [14]:
chinstrap_df = bird[bird["species"] == "Chinstrap"]

In [15]:
chinstrap_df["body_mass_g"].mean()

np.float64(3733.0882352941176)

<h2>Cleaning the Data Frame</h2>

In [16]:
bird.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>How do we check for NULL VALUES</h2>

In [17]:
bird["bill_length_mm"].mean()

np.float64(43.9219298245614)

In [18]:
bird["bill_depth_mm"].mean()

np.float64(17.151169590643274)

In [19]:
bird["flipper_length_mm"].mean()

np.float64(200.91520467836258)

In [20]:
bird["body_mass_g"].mean()

np.float64(4201.754385964912)

In [21]:
bird["sex"].mode()

0    male
Name: sex, dtype: object

<h2>How to Fill-up NULL VALUES</h2>

In [22]:
bird["bill_length_mm"] = bird["bill_length_mm"].fillna(bird["bill_length_mm"].mean())
bird["bill_depth_mm"] = bird["bill_depth_mm"].fillna(bird["bill_depth_mm"].mean())
bird["flipper_length_mm"] = bird["flipper_length_mm"].fillna(bird["flipper_length_mm"].mean())
bird["body_mass_g"] = bird["body_mass_g"].fillna(bird["body_mass_g"].mean())
bird["sex"] = bird["sex"].fillna(bird["sex"].mode()[0])

In [23]:
bird.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Data Types</h2>

In [24]:
bird.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [25]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     344.000000         344.000000   344.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.443643       1.969027          14.020657   799.613058   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.250000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h2>Changing Data Types</h2>

In [26]:
bird["species"] = bird["species"].astype("category")
bird["year"] = bird["year"].astype("category")

In [27]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

<hr>
<h3>Exploratory Data Analysis (EDA)</h3>
<hr>

<h4>Which Species of Penguins has the most weight?</h4>

In [28]:
bird[bird["species"] == "Adelie"]["body_mass_g"].mean()

np.float64(3703.95891043398)

In [29]:
bird[bird["species"] == "Gentoo"]["body_mass_g"].mean()

np.float64(5068.965761177136)

In [30]:
bird[bird["species"] == "Chinstrap"]["body_mass_g"].mean()

np.float64(3733.0882352941176)

<h4>Insights:</h4>

<p>According to the result above, <strong>Gentoo</strong> is the species that has the most weight.</p> 
<hr>

<h4>What year did the penguins gain the most weight?</h4>

In [31]:
bird[bird["year"] == 2007]["body_mass_g"].mean()

np.float64(4125.24322169059)

In [32]:
bird[bird["year"] == 2008]["body_mass_g"].mean()

np.float64(4266.666666666667)

In [33]:
bird[bird["year"] == 2009]["body_mass_g"].mean()

np.float64(4210.222953216375)

<h4>Insights:</h4>

<h5>Based on the reult above, penguins born or alive in year <strong>2008</strong> have gain a lot of weight</h5>
<hr>

<h4>What species of penguins in year 2008 has most weight?</h4>

In [61]:
weight2008 = bird.groupby(["species", "year"])["body_mass_g"].mean()

In [62]:
weight2008

species    year
Adelie     2007    3706.535088
           2008    3742.000000
           2009    3664.903846
Chinstrap  2007    3694.230769
           2008    3800.000000
           2009    3725.000000
Gentoo     2007    5070.588235
           2008    5019.565217
           2009    5119.358054
Name: body_mass_g, dtype: float64

<h4>Insights:</h4>

<h5>According to the result, the species of penguins that has the most weight in year <strong>2008</strong> is the Gentoo, with and average body mass of 5019.56 which is the highest among the species</h5>
<hr>

<h4>What island is more prone for penguins species to have more weight?</h4>

In [70]:
islandweight = bird.groupby(["island"])["body_mass_g"].mean()

In [71]:
islandweight

island
Biscoe       4712.956871
Dream        3712.903226
Torgersen    3715.899123
Name: body_mass_g, dtype: float64

<h5>Based on the result that I achieved is that <strong>Biscoe</strong>, an island was more prone for penguins to have more weight. Additionally, Biscoe has a large amount of penguins</h5> <hr>

<h4>Which sex has the maximum flipper length?</h4>

In [74]:
sexflipper = bird.groupby(["sex"])["flipper_length_mm"].max()

In [75]:
sexflipper

sex
female    222.0
male      231.0
Name: flipper_length_mm, dtype: float64

<h4>Insight:</h4>

<h5>According to the result above, Male penguin had the longest flipper with the maximum length of <strong>231.0</strong> while the female penguins only has <strong>222.0</strong></h5><hr>


<h4>Which island is the most populated by penguins?</h4>

In [8]:
bird["island"].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h4>Insight:</h4>

<h5>Based on the result, Biscoe is the most populated island with a population of <strong>168</strong> while Dream had a population of <strong>123</strong> and Torgersen with <strong>52</strong></h5> <hr>

<h4>What species are the most dominant in year 2008?</h4>

In [16]:
bird[bird["year"] == 2008]["species"].value_counts()

species
Adelie       50
Gentoo       46
Chinstrap    18
Name: count, dtype: int64

<h5>According to the result, <strong>Adelie</strong> are the most dominant species in year 2008, followed by <strong>Gentoo</strong> and <strong>Chinstrap</strong></h5>

<hr>
<h3>Data Aggregation</h3>
<Hr></Hr>

In [38]:
sp_is_x = bird.groupby(["species", "island", "sex"]).size()

In [39]:
sp_is_x

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64